### DISCUSSION TITLE
[New User] Reading and writing IMU sensor (MPU-6050) data via Arduino

### DISCUSSION THREAD

**Discussion by ShubhoB** [bonsai-rx/bonsai#1608] [Q&A] [answered]
Hi everyone,

I am fairly new in using Bonsai and Arduino so please don't mind the basic question(s) I am going to ask!

My project involves acquiring data from 3 cameras and the IMU chip (only the gyroscope and accelerometer readings). Although the camera workflows are working, I have been unable to get legitimate readings from the IMU chip and saving it as a .csv file. While the sketch uploaded in the Arduino gives me the correct readings in the serial monitor, I can't get the same using CreateSerialPort (perhaps this is the wrong strategy to start with?). I am attaching the 2 scripts I tested for the Arduino (which I found online) and the bonsai workflow for the IMU part here.
Would be extremely helpful if one can nudge me in the right direction here.

Thanks a lot,
Subhodeep
[imu.zip](https://github.com/bonsai-rx/bonsai/files/13242845/imu.zip)

**Comment by bruno-f-cruz** [comment, marked as answer]
Disconnect the `CreateSerialPort` from the `ReadLine` operator and try again. If that doesnt work let us know!

**Reply by ShubhoB** [reply]
That worked! Thanks!
